R/model.R

```diff
@@ -1362,31 +1426,125 @@ artnum_mf <- function(calendar_quarter, art_number, naomi_mf) {
 
     if (out_quarter_id %in% dat$quarter_id) {
       dat <- dplyr::filter(dat, quarter_id == out_quarter_id)
+      #  Tag row selected as model inputs
+      dat$data_type <- "raw_included"
     } else {
     dat <- dat %>%
       dplyr::group_by(area_id, sex, age_group) %>%
       dplyr::summarise(min_data_quarter = min(quarter_id),
                        max_data_quarter = max(quarter_id),
-                       art_current = approx(quarter_id, art_current, out_quarter_id, rule =2)$y) %>%
-      dplyr::ungroup() %>%
+                       art_current = approx(quarter_id, art_current, out_quarter_id, rule =2)$y,
+                       .groups = "drop") %>%
       dplyr::filter(out_quarter_id > min_data_quarter - 4L,
                     out_quarter_id <= max_data_quarter)
+    #  Tag row selected as model inputs
+    dat$data_type <- "interpolated_included"
     }
 
     if (nrow(dat) == 0) {
       stop(t_("NO_ART_DATA_FOR_QUARTER",
               list(calendar_quarter = calendar_quarter)))
     }
 
-    artnum_dat <- dat %>%
+    #  Tag rows not selected as model inputs
+    cols <- c("area_id", "sex", "age_group" , "art_current", "calendar_quarter")
+    dat$calendar_quarter <- calendar_quarter
+
+    excluded <- dplyr::anti_join(
+      art_number %>% dplyr::select(dplyr::all_of(cols)),
+      dat %>% dplyr::select(dplyr::all_of(cols)),
+      by = cols)
+
+    excluded$data_type <- "raw_excluded"
+
+
+    all_dat <- rbind(dat %>% dplyr::select(all_of(cols), data_type), excluded)
+
+    # Process selected data inputs as model inputs
+    model_input <- dat %>%
       dplyr::transmute(
                area_id,
                sex,
                age_group,
                artnum_idx = dplyr::row_number(),
                art_current
              )
+
+    artnum_dat_out <- list(raw_input = all_dat,
+                           model_input = model_input)
+
   }
 
-  artnum_dat
+  artnum_dat_out
+
+}
+
+
+
+#' @param naomi_data Naomi object of class "naomi_data" and "naomi_mf"
+#'
+#' @details
+#'
+
+#'
+#' @export
+
+# YOUAREHERE
+
+extract_naomi_inputs <- function(naomi_data){
```

Comment:
This function takes `naomi_data` and formats it for the `output_zip`? So that then you can use this in the comparison report? If so I wonder if it belongs in `R/outputs.R` more?

Reply:
Agreed - this has been moved to outputs function